bug-fix/fix-details-gallery-and-compines

ahmedNaser7 opened the pull request:
Refactor: Migrate LazyColumn to LazyVerticalGrid in TvShowDetailsScreen

Refactor: Migrate ProductionCompanySection and GallerySection to LazyGridScope

refactors the `ProductionCompanySection` and `GallerySection` composables to be extensions on `LazyGridScope`. This change simplifies their implementation by leveraging the `items` builder within a `LazyVerticalGrid` for displaying production companies and gallery images respectively.

Key changes:
- `ProductionCompanySection` now uses `LazyGridScope.items` to display `CompanyProductionCard`s.
- `GallerySection` now uses `LazyGridScope.items` to display `AsyncImage`s.
- The previous manual row and column layout logic within these sections has been removed.
- Adjusted padding and size modifiers for items within the lazy grid.
- Updated previews to reflect the new `LazyGridScope` usage.

refactors the `TvShowDetailsScreen` to use `LazyVerticalGrid` instead of `LazyColumn` for its main content area.

This change allows for a more flexible layout, particularly for displaying items like gallery images and media cards in a grid format.

Key changes:
- Replaced `LazyColumn` with `LazyVerticalGrid`.
- Updated `item` calls to use `item(span = {GridItemSpan(maxLineSpan)})` to ensure full-width spans for sections that should not be part of the grid.
- Adapted `TvTopComponent` to accept `LazyGridState` instead of `LazyListState`.
- Adjusted how gallery and production company sections are displayed within the grid.

ahmedNaser7 committed:
Refactor: Migrate ProductionCompanySection and GallerySection to LazyGridScope

This commit refactors the `ProductionCompanySection` and `GallerySection` composables to be extensions on `LazyGridScope`. This change simplifies their implementation by leveraging the `items` builder within a `LazyVerticalGrid` for displaying production companies and gallery images respectively.

Key changes:
- `ProductionCompanySection` now uses `LazyGridScope.items` to display `CompanyProductionCard`s.
- `GallerySection` now uses `LazyGridScope.items` to display `AsyncImage`s.
- The previous manual row and column layout logic within these sections has been removed.
- Adjusted padding and size modifiers for items within the lazy grid.
- Updated previews to reflect the new `LazyGridScope` usage.

ahmedNaser7 committed:
Refactor: Update ProductionCompanySection preview to use LazyVerticalGrid

The preview for the `ProductionCompanySection` composable has been updated to utilize `LazyVerticalGrid`. This change ensures the preview accurately reflects the component's intended usage within a scrollable grid layout.

ahmedNaser7 committed:
Refactor: Migrate LazyColumn to LazyVerticalGrid in MovieDetailsScreen

This commit refactors the `MovieDetailsScreen` to use `LazyVerticalGrid` instead of `LazyColumn` for displaying content.

Key changes:
- Replaced `LazyColumn` with `LazyVerticalGrid`.
- Adapted item spans for different content sections (Description, Cast, Chips, etc.) to ensure proper layout within the grid.
- Updated `rememberLazyListState` to `rememberLazyGridState`.
- Adjusted the `gallerySection` and `productionCompanySection` to be composable functions that can be used directly within the grid's item scope.
- Modified `MovieTopComponent` to accept `LazyGridState` instead of `LazyListState`.

This change aims to improve the layout flexibility and potentially the visual presentation of the movie details.

ahmedNaser7 committed:
Refactor: Migrate LazyColumn to LazyVerticalGrid in TvShowDetailsScreen

This commit refactors the `TvShowDetailsScreen` to use `LazyVerticalGrid` instead of `LazyColumn` for its main content area.

This change allows for a more flexible layout, particularly for displaying items like gallery images and media cards in a grid format.

Key changes:
- Replaced `LazyColumn` with `LazyVerticalGrid`.
- Updated `item` calls to use `item(span = {GridItemSpan(maxLineSpan)})` to ensure full-width spans for sections that should not be part of the grid.
- Adapted `TvTopComponent` to accept `LazyGridState` instead of `LazyListState`.
- Adjusted how gallery and production company sections are displayed within the grid.

ahmedNaser7 committed:
Update version to 1.0.19 and fix details gallery/company issue

ahmedNaser7 committed:
Refactor: Remove unused LazyListState import

The import of `LazyListState` was removed from `TvTopComponent.kt` and `MovieTopComponent.kt` as it was not being used.

Joseph-Sameh-0 commented on feature/mediaDetails/mediaDetailsUi/src/main/java/com/feature/mediaDetails/mediaDetailsUi/ui/screen/movie/details/MovieDetailsScreen.kt:
```diff
@@ -234,20 +236,21 @@ fun MovieDetailsScreenContent(
                     ) {
                         if (state.isDescriptionLoading) {
-                            item {
+                            item (span = {GridItemSpan(1)}){
```
```suggestion
                            item (span = {GridItemSpan(maxLineSpan)}){
```

Joseph-Sameh-0 committed:
Update feature/mediaDetails/mediaDetailsUi/src/main/java/com/feature/mediaDetails/mediaDetailsUi/ui/screen/movie/details/MovieDetailsScreen.kt

Joseph-Sameh-0 commented on feature/mediaDetails/mediaDetailsUi/src/main/java/com/feature/mediaDetails/mediaDetailsUi/ui/screen/tvShow/details/TvShowDetailsScreen.kt:
```diff
@@ -418,37 +420,39 @@ fun TvShowDetailsScreenContent(
-                                                style = Theme.textStyle.label.large,
-                                                color = Theme.colors.text.body.copy(alpha = 0.6f)
-                                            )
+                                        item {
```
```suggestion
                                        item (span = {GridItemSpan(maxLineSpan)}){
```

ahmedNaser7 committed:
Fix: Description placeholder span on media details screens

The `PageLoadingPlaceHolder` for the description on the MovieDetailsScreen and TvShowDetailsScreen was previously not spanning the full width.

This commit updates the `GridItemSpan` for the placeholder to use `maxLineSpan` to ensure it spans the full width of the screen.

ahmedNaser7 committed:
Merge remote-tracking branch 'origin/bug-fix/fix-details-gallery-and-compines' into bug-fix/fix-details-gallery-and-compines

ahmedNaser7 committed:
Merge branch 'develop' into bug-fix/fix-details-gallery-and-compines
